dnf-docker-test/features/steps/command_utils.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import codecs
+import shlex
+import subprocess
+
+import six
+
+__all__ = ["run"]
+
+class CommandResult(object):
+    def __init__(self, **kwargs):
+        self.command = kwargs.pop("command", None)
+        self.returncode = kwargs.pop("returncode", 0)
+        self.stdout = kwargs.pop("stdout", "")
+        self.stderr = kwargs.pop("stderr", "")
+        if kwargs:
+            names = ", ".join("{!r}".format(x) for x in kwargs)
+            raise ValueError("Unexpected: {!s}".format(names))
+
+    @property
+    def failed(self):
+        return self.returncode != 0
+
+    def clear(self):
+        self.command = None
+        self.returncode = 0
+        self.stdout = ""
+        self.stderr = ""
+
+class Command(object):
+    COMMAND_MAP = {}
+
+    @classmethod
+    def run(cls, command, **kwargs):
+        """
+        Make a subprocess call, collect its output and return code.
+        """
+        assert isinstance(command, six.string_types)
+        cmd_result = CommandResult()
+        cmd_result.command = command
+
+        if six.PY2 and isinstance(command, six.text_type):
+            # In PY2, shlex.split() requires bytes string (non-unicode).
+            # In PY3, shlex.split() accepts unicode string.
+            command = codecs.encode(command, "utf-8")
+        cmdargs = shlex.split(command)
+
+        command0 = cmdargs[0]
+        real_command = cls.COMMAND_MAP.get(command0, None)
+        if real_command:
+            cmdargs0 = real_command.split()
+            cmdargs = cmdargs0 + cmdargs[1:]
```

Comment:
Just see how many steps you perform before you run command by subproccess? Is it really worth to do them?